pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_drop_column.go

```diff
@@ -353,7 +353,18 @@ func walkDropColumnDependencies(b BuildCtx, col *scpb.Column, fn func(e scpb.Ele
 			} else if elt.EmbeddedExpr != nil {
 				for _, columnID := range elt.EmbeddedExpr.ReferencedColumnIDs {
 					if columnID == col.ColumnID {
-						fn(e)
+						indexElts := b.QueryByID(elt.TableID).Filter(hasIndexIDAttrFilter(elt.IndexID))
```

Comment:
maybe add a helper function like this to help (assuming that we only want to panic if the index predicate references the column, and it's ok that the index columns contain the column):
```
func maybePanicOnReferencingPartialIndex(b BuildCtx, col *scpb.Column, tblElts ElementResultSet) {
	var referencingIndexes catid.IndexSet
	tblElts.ForEachElementStatus(func(_ scpb.Status, _ scpb.TargetStatus, e scpb.Element) {
		switch elt := e.(type) {
		case *scpb.SecondaryIndex:
			if elt.EmbeddedExpr != nil {
				for _, columnID := range elt.EmbeddedExpr.ReferencedColumnIDs {
					if columnID == col.ColumnID {
						referencingIndexes.Add(elt.IndexID)
					}
				}
			}
		case *scpb.SecondaryIndexPartial:
			for _, columnID := range elt.ReferencedColumnIDs {
				if columnID == col.ColumnID {
					referencingIndexes.Add(elt.IndexID)
				}
			}
		}
	})

	if referencingIndexes.Len() > 0 {
		indexNames := make([]string, 0, referencingIndexes.Len())
		var colName string
		tblElts.ForEachElementStatus(func(_ scpb.Status, _ scpb.TargetStatus, e scpb.Element) {
			switch elt := e.(type) {
			case *scpb.IndexName:
				if referencingIndexes.Contains(elt.IndexID) {
					indexNames = append(indexNames, elt.Name)
				}
			case *scpb.ColumnName:
				if elt.ColumnID == col.ColumnID {
					colName = elt.Name
				}
			}
		})
		panic(errors.WithHint(
			pgerror.Newf(
				pgcode.InvalidColumnReference,
				"column %q cannot be dropped because it is referenced by partial index (%v)",
				colName, indexNames,
			),
			"drop the partial index first, then drop the column",
		))
	}
}
```